frontend/src/crosswordView/crossword.js

```diff
@@ -0,0 +1,114 @@
+            if (grid[curR][curC] !== '#') {
+                this.cellRefs[`cell-${curR},${curC}`].focus();
+                break;
+            }
+            curR = (gridHeight + curR + dr) % gridHeight;
+            curC = (gridWidth + curC + dc) % gridWidth;
+        }
+        this.setState({ focusRow: curR, focusCol: curC });
+    }
+
+    handleKeyDown = (e) => {
+        const deltas = {
+            ArrowLeft: [0, -1],
+            ArrowRight: [0, 1],
+            ArrowUp: [-1, 0],
+            ArrowDown: [1, 0],
+        };
+        if (['ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'].includes(e.code)) {
+            e.preventDefault();
+            const { code } = e;
+            this.changeFocusDelta(deltas[code][0], deltas[code][1]);
+        }
+    }
+
+    changeFocus = (r, c) => {
+        this.cellRefs[`cell-${r},${c}`].focus();
+        this.setState({ focusRow: r, focusCol: c });
+    }
+
+    render() {
+        return this.props.grid.map((row, r) => {
+            return (
+                <div className='crossword-row' key={r}>
+                    {row.map((cell, c) => {
+                        if (cell === '#') {
+                            return (
+                                <div className='blank-cell' key={c}>
+                                    <input
+                                        className='blank-input'
+                                        readOnly
+                                    />
+                                </div>
+                            );
+                        }
+                        const clueNum = this.props.getClueNumber(r, c);
+                        const isCorrect = this.props.isCorrect(r, c);
+                        return (
+                            <div className='crossword-cell' key={c}>
+                                <div className='clue-num'>
+                                    {clueNum}
+                                </div>
+                                <input
+                                    className={`cell-input ${isCorrect ? 'correct-cell' : ''}`}
+                                    type='text'
+                                    value={cell}
+                                    maxLength={1}
+                                    ref={(instance) => {
+                                        this.cellRefs[`cell-${r},${c}`] = instance;
+                                    }}
+                                    onClick={() => this.changeFocus(r, c)}
+                                    onChange={(e) => this.props.updateCell(e, r, c)}
+                                />
+                            </div>
+                        );
+                    })}
+                </div>
+            );
+        });
+    }
+}
+Crossword.propTypes = {
+    grid: PropTypes.array,
+    updateCell: PropTypes.func,
+    getClueNumber: PropTypes.func,
+    isCorrect: PropTypes.func,
+};
+
+export default Crossword;
```

Comment:
I think this component is very well constructed, great work! I appreciate the consistency in the use of variables names throughout this component, though I'd encourage everyone not to be afraid to expand on variable names for the sake of readability. It looks like `r` and `c` in this component are consistently used to refer to something like `row_index` and `column_index`, respectively. It'd probably help our future selves out to expand those variable names to make it clear in everyone context what they refer to. One example: there's a little inconsistency there when we pass up those index references to `this.props.updateCell` in the parent component, which refers to them as `row` and `col`, respectively. I think it'd be worth it to take a quick pass at both this and `crosswordView` to ensure that we're consistent in how we refer to these values.